datastore/api/snippets.py

```diff
@@ -0,0 +1,820 @@
+# Copyright 2016, Google, Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from collections import defaultdict
+import datetime
+from pprint import pprint
+import time
+
+import gcloud
+from gcloud import datastore
+
+
+def incomplete_key(client):
```

Comment:
Seems kind of silly to pass the same client to every single method. But I also understand the aversion to a module level variable or classes. idk, just a thought.